torchao/quantization/quant_primitives.py

```diff
@@ -258,9 +336,9 @@ def dequantize_affine(
     if zero_point_domain == ZeroPointDomain.INT:
         # Force a copy to avoid input modification due
         # to upcoming in-place operations.
-        dequant = input.to(torch.int32, copy=True)
+        dequant = input.clone()
         if zero_point is not None:
-            dequant -= zero_point.to(torch.int32)
+            dequant -= zero_point
         dequant = dequant.to(output_dtype)
```

Comment:
we have type casting here as well, this casting should probably happen early I think